### ISSUE TITLE
strpos(): Empty needle in Parser.php on line 557

### ISSUE THREAD

**Issue by JanPetterMG** [simplepie/simplepie#529]
`Warning: strpos(): Empty needle in /library/SimplePie/Parser.php on line 557`

I'm aware that this library is abandoned, but does anyone have a clue why this warning shows up?
It happens every minute, for multiple feeds.

https://github.com/simplepie/simplepie/blob/master/library/SimplePie/Parser.php#L557

Is it an idea to change
```php
$photo_list = array();
for ($j = 0; $j < count($entry['properties']['photo']); $j++) {
	$photo = $entry['properties']['photo'][$j];
	if (strpos($content, $photo) === false) {
		$photo_list[] = $photo;
	}
}
```
to include `!empty($photo)` in the `if`
```php
$photo_list = array();
for ($j = 0; $j < count($entry['properties']['photo']); $j++) {
	$photo = $entry['properties']['photo'][$j];
	if (!empty($photo) &&
	    strpos($content, $photo) === false
	) {
		$photo_list[] = $photo;
	}
}
```
Or does anyone have a better solution?

**Comment by skyzyx** [comment]
What's the test feed?

**Comment by JanPetterMG** [comment]
Some random feeds, all in RDF format, and all Sweedish :/
http://www.blt.se/blekinge/feed/rdf/
http://www.barometern.se/hogsby/feed/rdf/
http://www.barometern.se/kalmar/feed/rdf/
http://www.barometern.se/monsteras/feed/rdf/
http://www.barometern.se/oskarshamn/feed/rdf/

To reproduce you'll need the `mf2/mf2` library.
I'm at `v0.3.2` (latest), released about 2 months ago. I did some updates about a week ago, so it might be related.
For info, I've been using `v0.3.0` ever since I discovered this Simplepie+mf2 integration about a year ago.
Should I switch to an older version, eg. `v0.2.x` or `v0.1.x`? There isn't much documentation about Simplepie and mf2 around...

**Comment by mblaney** [comment]
Hi @JanPetterMG original SimplePie hasn't been abandoned, at least not yet ;-)

I didn't realise anyone else was using SimplePie+mf2, I have a few patches for it that I should push. It is possible that the mf2 parser could return an empty string, so your additional check makes sense. (I would be putting some brackets around the second condition in the if statement though.)

**Comment by mblaney** [comment]
closed via #530
